### PR TITLE
Issue #9768: updated RightCurlyCheckTest to use unique Input file in …

### LINKED ISSUE #9768: update RightCurlyCheckTest to use unique Input file in each test method
RightCurlyCheckTest.java file reuses some Input files few times in few different test method.
Expected: each test method in Test class should use its own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
Closes #9768 
Made minor changes and created unique Input files for each method.

### CODE REVIEW

**Comment by Devansu-Yadav** [comment]
@timurt @romani Can u pls review my PR? Also can u pls approve #9768 to avoid any CI failures?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestSame.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        int x = 1; // should not require any javadoc
+    }
+
+
+
+    public enum GreetingsEnum
+    {
+        HELLO,
+        GOODBYE
+    }; // ok
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestSame.java:84]: it says ok, but violations are expected according to the test case, please check also lines 91, 186, and 187

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestNewLine.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        int x = 1; // should not require any javadoc
+    }
+
+
+
+    public enum GreetingsEnum
+    {
+        HELLO,
+        GOODBYE
+    }; // ok
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestNewLine.java:84]: violation here, please check test case

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestShouldStartLine2.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        }; // ok
+        for (int i = 0; i < 1; i++) { new Object(); }; // ok
+    }
+
+    static final java.util.concurrent.ThreadFactory threadFactory
+            = new java.util.concurrent.ThreadFactory() {
+        @Override
+        public Thread newThread(final Runnable r) {
+            return new Thread(r);
+        }}; // ok
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestShouldStartLine2.java:194]: 194 and 199 lines violations

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestShouldStartLine2.java (hunk 3 above)
Comment (Devansu-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestShouldStartLine2.java]: Done!

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestNewLine.java (hunk 2 above)
Comment (Devansu-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestNewLine.java]: Done!

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestSame.java (hunk 1 above)
Comment (Devansu-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeftTestSame.java]: Done!

**Review by romani — APPROVED** [review]

**Pull request merged**